build/deploy/Dockerfile

```diff
@@ -1,28 +1,26 @@
-FROM debian:9.12-slim
+FROM registry.access.redhat.com/ubi8/ubi
```

Comment:
Can we use `ubi-minimal` instead of the regular `ubi`? It looks like all the packages we list below are present there (https://catalog.redhat.com/software/containers/ubi8/ubi-minimal/5c359a62bed8bd75a2c3fba8?container-tabs=packages).

Reply:
I had tried `ubi-minimal`. Cockroach couldn't find the installed tzinfo. I verified the package was installed but didn't dig deeper at the time. I just dug a bit deeper and found that the minimal image gets it's smaller size by excluding documentation and language files for many RPMs. For tzinfo, all the files in `/usr/share/zoneinfo` are excluded.

Reply:
Oops. Sounds like a bug on redhat's side, but when this was reported in `rhel-atomic` they responded by removing the rest of the tzdata package instead of fixing it. It looks like it's *intended* to be there since it still shows up in the package list, but the actual contents are missing.

Is there any way to use ubi-minimal and then reinstall the tzdata package? It looks like `yum` is not present on ubi-minimal and I don't know enough of the redhat ecosystem to know if there's an alternative that is present.

Reply:
The tool to use in `ubi-minimal` is `microdnf`. [This is the primary page](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/building_running_and_managing_containers/using_red_hat_universal_base_images_standard_minimal_and_runtimes) I've been referencing for information on what's in each of the UBI image versions and how they differ from each other. The "Based on RHEL packaging" bullet of [section 4.1.2](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/building_running_and_managing_containers/using_red_hat_universal_base_images_standard_minimal_and_runtimes#using_minimal_red_hat_base_images) on that page is where I found the note about what is excluded from the `ubi-minimal` image. The command `rpm -Va` shows the excluded files for the installed RPMs.

I tried uninstalling the tzdata package and found it is depended upon by a large number of packages. There doesn't appear to be a way to force uninstallation and I'm not finding a way to install a different version of the file.

It does seem odd that they'd intentionally include a package and remove all the useful information from it. I can put a bug report in for this.

Reply:
I figured out a hack to make the image use the `ubi8/ubi-minimal` image. See https://github.com/cockroachdb/cockroach/pull/55467.

Reply:
This issue says that there's now a `microdnf reinstall` command: https://github.com/rpm-software-management/microdnf/issues/34

And if that's too recent, it offers a workaround: `rpm --erase --nodeps tzdata && microdnf install tzdata`

Reply:
Ah, I see you're already using `microdnf reinstall` in #55467.

Reply:
Oh. Nice find. I'm switching to use the rpm erase hack.